internal/servers/servers_test.go

```diff
@@ -274,30 +290,46 @@ func TestShutdown(t *testing.T) {
 					}
 					return nil
 				},
+				afterFunc: func() {
+					cancel()
+					eg.Wait()
+				},
 				want: nil,
 			}
 		}(),
-		{
-			name: "server not found error",
-			args: args{
-				ctx: context.Background(),
-			},
-			field: field{
-				eg:      errgroup.Get(),
-				servers: map[string]server.Server{},
-				sds: []string{
-					"srv1",
+		func() test {
+			ctx, cancel := context.WithCancel(context.Background())
+			eg, ctx := errgroup.New(ctx)
+
+			return test{
+				name: "server not found error",
+				args: args{
+					ctx: ctx,
 				},
-			},
-			checkFunc: func(got, want error) error {
-				if !errors.Is(want, got) {
-					return errors.Errorf("not equals. want: %v, got: %v", want, got)
-				}
-				return nil
-			},
-			want: errors.ErrServerNotFound("srv1"),
-		},
+				field: field{
+					eg:      eg,
+					servers: map[string]server.Server{},
+					sds: []string{
+						"srv1",
+					},
+				},
+				checkFunc: func(got, want error) error {
+					if !errors.Is(want, got) {
+						return errors.Errorf("not equals. want: %v, got: %v", want, got)
+					}
+					return nil
+				},
```

Comment:
could you define `defaultCheckFunc` instead of defining here?
It may be used for all cases.

Reply:
Thank you for your suggestion! It looks good!

I have fixed: https://github.com/vdaas/vald/pull/1689/commits/f20fba29e1983edd66155c8746c142bfd6e2a390